A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_PerformancePage.js

```diff
@@ -706,34 +707,45 @@ class CDUPerformancePage {
                 }
             }
         };
-        let titleColor = "white";
-        if (mcdu.currentFlightPhase === FlightPhase.FLIGHT_PHASE_APPROACH) {
-            titleColor = "green";
-        }
-        let qnhCell = "[ ]";
-        const QNH_REGEX = /[0-9]{2}.[0-9]{2}/;
-        if (isFinite(mcdu.perfApprQNH) || QNH_REGEX.test(mcdu.perfApprQNH)) {
-            qnhCell = mcdu.perfApprQNH;
+
+        let qnhCell = "[\xa0\xa0][color]cyan";
+        if (isFinite(mcdu.perfApprQNH)) {
+            if (mcdu.perfApprQNH < 500) {
+                qnhCell = mcdu.perfApprQNH.toFixed(2) + "[color]cyan";
+            } else {
+                qnhCell = mcdu.perfApprQNH.toFixed(0) + "[color]cyan";
+            }
+        } else if (mcdu.flightPlanManager.getDestination() && mcdu.flightPlanManager.getDestination().infos) {
+            const lat = SimVar.GetSimVarValue("PLANE LATITUDE", "degree latitude");
+            const long = SimVar.GetSimVarValue("PLANE LONGITUDE", "degree longitude");
+            const planeLla = new LatLongAlt(lat, long);
+            const distance = Avionics.Utils.computeGreatCircleDistance(
+                planeLla, mcdu.flightPlanManager.getDestination().infos.coordinates
+            );
+            if (distance <= 180) {
+                qnhCell = "____[color]amber";
+            }
         }
         mcdu.onLeftInput[0] = (value) => {
             if (mcdu.setPerfApprQNH(value)) {
                 CDUPerformancePage.ShowAPPRPage(mcdu);
             }
         };
-        let tempCell = "[ ]";
+
+        let tempCell = "[\xa0]";
         if (isFinite(mcdu.perfApprTemp)) {
-            tempCell = mcdu.perfApprTemp.toFixed(0);
+            tempCell = ("" + mcdu.perfApprTemp.toFixed(0)).padStart(3).replace(/ /g, "\xa0");
         }
         mcdu.onLeftInput[1] = (value) => {
             if (mcdu.setPerfApprTemp(value)) {
                 CDUPerformancePage.ShowAPPRPage(mcdu);
             }
         };
-        let magWindHeadingCell = "[ ]";
+        let magWindHeadingCell = "[\xa0]";
         if (isFinite(mcdu.perfApprWindHeading)) {
-            magWindHeadingCell = mcdu.perfApprWindHeading.toFixed(0);
+            magWindHeadingCell = ("" + mcdu.perfApprWindHeading.toFixed(0)).padStart(3, 0);
         }
-        let magWindSpeedCell = "[ ]";
+        let magWindSpeedCell = "[\xa0]";
         if (isFinite(mcdu.perfApprWindSpeed)) {
             magWindSpeedCell = mcdu.perfApprWindSpeed.toFixed(0);
```

Comment:
Looks like this should also be padLeft'ed, so that it's 001 (according to your small reference screenshot).